cmake/Googletest.cmake

```diff
@@ -79,4 +83,10 @@ macro(fetch_googletest)
         set(GTEST_MAIN_LIBRARIES ${GTEST_MAIN_LIBRARY})
     endif()
 
+    set(GTEST_LIBS ${GTEST_LIBRARIES})
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Windows" AND NOT CMAKE_BUILD_TYPE MATCHES Debug)
+        set(GTEST_LIBS "${GTEST_ROOT}/bin/gtest.dll")
+    endif()
+
+    install(FILES ${GTEST_LIBS} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/gtest)
```

Comment:
We now install gtest shared library into ${CMAKE_INSTALL_PREFIX}/lib/gtest that unit tests needs to link against.